### PR TITLE
added GaussianSampler, ResizeBilinear for VAE

### PR DESCRIPTION
**Pull request by weiguanghuang**
updated:
KerasLayerWrapper.scala

added:
GaussianSampler.scala
ResizeBilinear.scala
GaussianSamplerSpec.scala
ResizeBilinearSpec.scala

### CODE REVIEW

**Comment by weiguanghuang** [comment]
cc @zhichao-li

**Comment by zhichao-li** [comment]
Looks good, but you need to rebase the code before running jenkins verification.

File: pyzoo/zoo/pipeline/api/keras/layers/torch.py [hunk 1]
```diff
@@ -697,3 +697,29 @@ def __init__(self, value=1e-6, input_shape=None, **kwargs):
                                               float(value),
                                               list(input_shape) if input_shape else None,
                                               **kwargs)
+
+
+class GaussianSampler(ZooKerasLayer):
+    """
+    >>> gaussianSampler = GaussianSampler(input_shape=(2, 3, 5, 7))
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras/layers/torch.py]: Remember to add doc for both scala and python :)

File: pyzoo/zoo/pipeline/api/keras/layers/torch.py [hunk 2]
```diff
@@ -697,3 +697,29 @@ def __init__(self, value=1e-6, input_shape=None, **kwargs):
...
+                                              **kwargs)
+
+
+class ResizeBilinear(ZooKerasLayer):
+    """
+    >>> resizeBilinear = ResizeBilinear(10, 20, False, "NCHW")
+    createZooKerasResizeBilinear
+    """
+    def __init__(self, output_height, output_width, align_corner=False,
+                 data_format="NCHW", **kwargs):
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras/layers/torch.py]: change to `dim_ordering`

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/GaussianSampler.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.{nn => bnn}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Shape
+import com.intel.analytics.zoo.pipeline.api.keras.layers.utils.KerasUtils
+
+import scala.reflect.ClassTag
+
+/**
+ * Takes {mean, log_variance} as input and samples from the Gaussian distribution
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/GaussianSampler.scala]: Enrich doc

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/GaussianSamplerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.{RandomGenerator, Shape, T, Table}
+import com.intel.analytics.zoo.pipeline.api.keras.ZooSpecHelper
+import com.intel.analytics.zoo.pipeline.api.keras.serializer.ModuleSerializationTest
+
+import scala.util.Random
+
+class GaussianSamplerSpec extends ZooSpecHelper {
+
+  "GaussianSampler (3, 1) Zoo" should "be the same as BigDL" in {
```
Comment (hkvision) [inline zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/GaussianSamplerSpec.scala]: what is (3, 1)?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras.scala [hunk 5]
```diff
@@ -993,4 +993,17 @@ class PythonZooKeras[T: ClassTag](implicit ev: TensorNumeric[T]) extends PythonB
...
+  def createZooKerasGaussianSampler(
+      inputShape: JList[Int] = null): GaussianSampler[T] = {
+    GaussianSampler(toScalaShape(inputShape))
+  }
+
+  def createZooKerasResizeBilinear(
+      outputHeight: Int,
+      outputWidth: Int,
+      alignCorners: Boolean,
+      dimOrdering: String = "th"): ResizeBilinear[T] = {
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/python/PythonZooKeras.scala]: inputShape?

File: pyzoo/zoo/pipeline/api/keras/layers/torch.py [hunk 6]
```diff
@@ -697,3 +697,51 @@ def __init__(self, value=1e-6, input_shape=None, **kwargs):
...
+
+    # Arguments
+    output_height: output height
+    output_width: output width
+    align_corner: align corner or not
+    dim_ordering: Format of input data. Either 'th' (Channel First) or 'tf' (Channel Last).
+                  Default is 'th'.
+    input_shape: A shape tuple, not including batch.
+
+    >>> resizeBilinear = ResizeBilinear(10, 20)
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/keras/layers/torch.py]: input_shape?

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/475/

**Comment by jason-dai** [comment]
@wngzhiqi @yangw1234 we can now port the VAE notebook to Keras APIs?

**Pull request merged**